### ISSUE TITLE
Cannot retrive dataset metadata 

### ISSUE THREAD

**Issue by menegon** [ioos/thredds_crawler#23]
thredds_crawler: v. 1.5.3
lxml: v. 3.6.2.

If I try to reproduce the code in the documentation: 

`from thredds_crawler.crawl import Crawl`
`c = Crawl('http://tds.maracoos.org/thredds/MODIS.xml', select=['.*-Agg'])`
`dataset = c.datasets[0]`
`print dataset.metadata.find("{http://www.unidata.ucar.edu/namespaces/thredds/InvCatalog/v1.0}documentation").text`

I get this error:

```
AssertionError                            Traceback (most recent call last)
<ipython-input-4-6b79b125cab6> in <module>()
----> 1 print dataset.metadata.find("{http://www.unidata.ucar.edu/namespaces/thredds/InvCatalog/v1.0}documentation").text

src/lxml/lxml.etree.pyx in lxml.etree._Element.find (src/lxml/lxml.etree.c:57129)()

/home/ckan/.virtualenvs/sp7-geogate/local/lib/python2.7/site-packages/lxml/_elementpath.pyc in find(elem, path, namespaces)
    293             return next(it)
    294         else:
--> 295             return _next()
    296     except StopIteration:
    297         return None

/home/ckan/.virtualenvs/sp7-geogate/local/lib/python2.7/site-packages/lxml/_elementpath.pyc in select(result)
     87     def select(result):
     88         for elem in result:
---> 89             for e in elem.iterchildren(tag):
     90                 yield e
     91     return select

src/lxml/lxml.etree.pyx in lxml.etree._Element.iterchildren (src/lxml/lxml.etree.c:56171)()

src/lxml/lxml.etree.pyx in lxml.etree.ElementChildIterator.__cinit__ (src/lxml/lxml.etree.c:73465)()

src/lxml/apihelpers.pxi in lxml.etree._assertValidNode (src/lxml/lxml.etree.c:15151)()


AssertionError: invalid Element proxy at 140187486574496
```

I get the same error when I try to retrieve the metadata from another server.

**Comment by huard** [comment]
Same here with python2. Looks similar to 
https://stackoverflow.com/questions/29570715/how-to-fix-lxml-assertion-error

With python3: 
```
MaybeEncodingError: Error sending result: '<LeafDataset id: birdhouse/ncep/narr/snod/snod.2010.nc, name: snod.2010.nc, services: ['HTTPServer', 'OPENDAP']>'. Reason: 'TypeError("can't pickle lxml.etree._Element objects",)'
```

**Comment by andrmos** [comment]
I had similar issues with thredds_crawler: v. 1.5.3. Version 1.5.4 fixed the issues for me.

**Comment by kwilcox** [comment]
Yes, this is fixed in 1.5.4!
